src/Renci.SshNet/Security/KeyExchangeECDH.cs

```diff
@@ -55,7 +55,7 @@ public override void Start(Session session, KeyExchangeInitMessage message, bool
                 return;
             }
 #endif
-            var curveParameter = SecNamedCurves.GetByName(CurveName);
```

Comment:
As far as I can tell, going through SshNamedCurves causes a bunch of extra lookup and static initialization compared to SecNamedCurves. Is there any reason to use it?

It's a shame we can't just directly refer to the static BC object when we know the curve we want.

Reply:
I guess `SshNamedCurves` is designed for libraries like SSH.NET to consume?

Reply:
Maybe, but I don't see why we would. If I run this program in a loop:

```c#
public static void Main()
{
    var s = Stopwatch.StartNew();

    //if (!SecNamedCurves.GetByName("secp256r1").G.IsInfinity)
    if (!SshNamedCurves.GetByName("nistp256").G.IsInfinity)
    {
        Console.WriteLine(s.ElapsedTicks);
    }
}
```
```ps
dotnet build -c Release; do { dotnet run -c Release --no-build } while ($true)
```

The result is about 28.3±0.7ms vs 29.2±0.8ms. Both extremely high (also includes first JIT of relevant BouncyCastle methods), but 1ms is 1ms

Reply:
Would you mind trying `SecNamedCurves.GetByOid(SecObjectIdentifiers.SecP256r1)`?

Reply:
I didn't try it but I'm pretty sure it would be the same as `SecNamedCurves.GetByName`. The problem with `SshNamedCurves` is the amount of static constructors that it forces to run